firewood/src/v2/api.rs

```diff
@@ -172,7 +174,7 @@ pub trait DbView {
 /// [DbView], which means you can fetch values from it or
 /// obtain proofs.
 #[async_trait]
-pub trait Proposal: DbView {
+pub trait Proposal: DbView + Send + Sync {
```

Comment:
There's nothing inherent about a proposal that should necessitate `Send` and `Sync` trait bounds. Something seems off here. It would be better if we could put extra bounds on the generic types at the location where `Send` and `Sync` are actually necessary. 

Unless `Send` and `Sync` are necessary for the actual trait implementation, but I'm not sure that that is the case, is it?

Reply:
```console
➜  firewood git:(main) gh pr checkout 323
Switched to branch 'rkuris/async-down-stack'
Your branch is up to date with 'origin/rkuris/async-down-stack'.
Already up to date.
➜  firewood git:(rkuris/async-down-stack) sd '(DbView) \+ Send \+ Sync' '$1' firewood/src/v2/api.rs
➜  firewood git:(rkuris/async-down-stack) ✗ git --no-pager diff
diff --git a/firewood/src/v2/api.rs b/firewood/src/v2/api.rs
index 0ddacbe..dbc9ff4 100644
--- a/firewood/src/v2/api.rs
+++ b/firewood/src/v2/api.rs
@@ -172,7 +172,7 @@ pub trait DbView {
 /// [DbView], which means you can fetch values from it or
 /// obtain proofs.
 #[async_trait]
-pub trait Proposal: DbView + Send + Sync {
+pub trait Proposal: DbView {
     type Proposal: DbView + Proposal;

     /// Commit this revision
➜  firewood git:(rkuris/async-down-stack) ✗ cargo check -p firewood
    Checking firewood v0.0.4 (/Users/richardpringle/code/firewood/firewood)
    Finished dev [unoptimized + debuginfo] target(s) in 0.40s
➜  firewood git:(rkuris/async-down-stack) ✗
```

FYI
This might not be the case for you, but I never remember how to use `sed`, I much prefer [`sd`](https://github.com/chmln/sd).

`ripgrep` is also much better than `grep` 😉

Reply:
Without this, you get this when you wrap the Proposal in an Arc:
```
usage of an `Arc` that is not `Send` or `Sync`
the trait `Send` is not implemented for `<impl Db as Db>::Proposal`
the trait `Sync` is not implemented for `<impl Db as Db>::Proposal`
required for `Arc<<impl Db as Db>::Proposal>` to implement `Send` and `Sync`
consider using an `Rc` instead or wrapping the inner type with a `Mutex`
for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#arc_with_non_send_sync
`#[warn(clippy::arc_with_non_send_sync)]` on by default
```